### PR TITLE
Consolidate changes from WPCOM's CSSTidy

### PR DESCRIPTION
**Pull request by fullofcaffeine**
#### Changes proposed in this Pull Request:

Follow up to: D74626-code and D79378-code.

The CSSTidy source code in Jetpack might be synced to other consumers, though. A good example is WPCOM. Unfortunately, it makes it easy to fork the source tree by accident, and that shouldn't happen, as CSSTidy is supposed to live in Jetpack - it's its source of truth.

This PR incorporates all the changes that were made directly in the WPCOM that weren't yet synced over. Luckily, there was only one change to transfer: D74626-code.

From now on, any changes to CSSTidy should be done in JP first and then synced over to WPCOM. Developers might directly change the WPCOM version in case of emergency, provided a PR to incorporate the changes in Jetpack is created shortly thereafter.

#### Testing instructions:

The changeset  here were already tested as part of D74626-code.

#### Does this pull request change what data or activity we track or use?

No.

### CODE REVIEW

File: projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php [hunk 1]
```diff
@@ -943,7 +943,11 @@ public function parse( $string ) {
 						}
 
 						if ( 'iv' === $this->status ) {
-							if ( ! $_quoted_string ) {
+							// phpcs:disable
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php:946]: If you specify the specific rules here and on line 960, then line 960 won't override line 32 and you won't have to add lines 990 and 1039.

Specifying the specific rules is also generally better practice IMO.

```php
// phpcs:disable Squiz.PHP.CommentedOutCode.Found, Squiz.Commenting.BlockComment.NoNewLine
...
// phpcs:enable Squiz.PHP.CommentedOutCode.Found, Squiz.Commenting.BlockComment.NoNewLine
```

File: projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php [hunk 2]
```diff
@@ -943,7 +943,11 @@ public function parse( $string ) {
...
+							// WPCOM hack: prevents CSSTidy from removing spaces after commans inside
+							// declaration's values.
+							// For more information, see https://code.a8c.com/D74626.
```
[the remark this answers is not in the document]
Comment (fullofcaffeine) [reply projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php:949]: Yeah, sorry, I totally missed that it's a public repo. Will rewrite the commit.

[the remark this answers is not in the document]
Comment (fullofcaffeine) [reply projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php:949]: Oh right, thanks for pointing that out!

[the remark this answers is not in the document]
Comment (fullofcaffeine) [reply projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php]: Fixed by amending and rewriting the history.

**Comment by fullofcaffeine** [comment]
@anomiex Thanks a lot for reviewing. This is ready for another look!

File: projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php [hunk 3]
```diff
@@ -943,7 +943,11 @@ public function parse( $string ) {
 						}
 
 						if ( 'iv' === $this->status ) {
-							if ( ! $_quoted_string ) {
+							// phpcs:disable Squiz.PHP.CommentedOutCode.Found, Squiz.Commenting.BlockComment.NoNewLine
+							// WPCOM hack: prevents CSSTidy from removing spaces after commans inside
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-css/csstidy/class.csstidy.php:947]:
```suggestion
							// WPCOM hack: prevents CSSTidy from removing spaces after commas inside
```

**Review by anomiex — COMMENTED** [review]
One typo in the comment, otherwise looks good to me.

**Comment by fullofcaffeine** [comment]
@anomiex Thanks, I've fixed the typo. I see the codecov check is failing? Can I just ignore it this time?

**Review by anomiex — DISMISSED** [review]
Yes, the codecov check can be ignored. You'll need to resolve the wpcom check though; p9dueE-wa-p2 can help there.

**Review by jeherve — DISMISSED** [review]

**Review by jeherve — APPROVED** [review]

**Pull request merged**